numba_rvsdg/core/datastructures/ast_transforms.py

```diff
@@ -655,11 +666,216 @@ def render(self) -> None:
         self.blocks.to_SCFG().render()
 
 
+class SCFG2ASTTransformer:
+
+    def transform(
+        self, original: ast.FunctionDef, scfg: SCFG
+    ) -> ast.FunctionDef:
+        body: list[ast.AST] = []
+        self.region_stack = [scfg.region]
+        self.scfg = scfg
+        for name, block in scfg.concealed_region_view.items():
+            if type(block) is RegionBlock and block.kind == "branch":
+                continue
+            body.extend(self.codegen(block))
+        fdef = ast.FunctionDef(
+            name="transformed_function",
+            args=original.args,
+            body=body,
+            lineno=0,
+            decorator_list=original.decorator_list,
+            returns=original.returns,
+        )
+        return fdef
+
+    def lookup(self, item: Any) -> Any:
+        subregion_scfg = self.region_stack[-1].subregion
+        parent_region_block = self.region_stack[-1].parent_region
+        if item in subregion_scfg:  # type: ignore
+            return subregion_scfg[item]  # type: ignore
+        else:
+            return self.rlookup(parent_region_block, item)  # type: ignore
+
+    def rlookup(self, region_block: RegionBlock, item: Any) -> Any:
+        if item in region_block.subregion:  # type: ignore
+            return region_block.subregion[item]  # type: ignore
+        elif region_block.parent_region is not None:
+            return self.rlookup(region_block.parent_region, item)
+        else:
+            raise KeyError(f"Item {item} not found in subregion or parent")
+
+    def codegen(self, block: Any) -> list[ast.AST]:
```

Comment:
This function can return malformed data; e.g. `[ [ [ node1, node2 ] ] ]`. The nested list can cause things like `ast.NodeTransformer` to misbehave. But things like `ast.unparse` works. I am current doing `ast.parse(ast.unparse(tree))` roundtrip to workaround the problem.

Reply:
yeah, it sometimes returns list that are nested rather than concatenated, I had been wondering why this works. Probably worth fixing.

Reply:
hopefully fixed in [3da4ece](https://github.com/numba/numba-rvsdg/pull/127/commits/3da4ecea0694d532e579748e16e907d73e15b21f)